src/compiler/checker.ts

```diff
@@ -16720,7 +16720,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function getStringMappingType(symbol: Symbol, type: Type): Type {
         return type.flags & (TypeFlags.Union | TypeFlags.Never) ? mapType(type, t => getStringMappingType(symbol, t)) :
             type.flags & TypeFlags.StringLiteral ? getStringLiteralType(applyStringMapping(symbol, (type as StringLiteralType).value)) :
-            type.flags & TypeFlags.TemplateLiteral ? getTemplateLiteralType(...applyTemplateStringMapping(symbol, (type as TemplateLiteralType).texts, (type as TemplateLiteralType).types)) :
+            type.flags & TypeFlags.TemplateLiteral && !isGenericType(type) ? getTemplateLiteralType(...applyTemplateStringMapping(symbol, (type as TemplateLiteralType).texts, (type as TemplateLiteralType).types)) :

```

Comment:
The problem in the repro is that this is a conditional type for which the falseType is replaced with the trueType (after checking the inner extendsType and checkType) but without instantiating~ the type param that was instantiated for the purpose of checking extendsType against checkType. So perhaps the alternative here is to do that? I could explore that solution if it’s the one that you’d prefer